eeps/eep-0057.md

```diff
@@ -0,0 +1,237 @@
+
+    -doc """
+    Encodes the given binary to base64.
+
+        1> base64:encode("hello").
+        <<"aGVsbG8=">>
+
+    """.
+    -spec encode(binary()) -> binary().
+    encode(Binary) ->
+      % ....
+
+And then in your test suite:
+
+    doctests(_Config) ->
+      ct_doctest:run(base64).
+
+The doctest attribute will access the documentation entries in the base64 Docs chunk, extract all of the examples, and run them. Of course, there is nothing stopping doctests from being implemented on top of EDoc today, but the fact the documentation is already parsed by the parser/compiler, makes doctests trivial to add. Doctests would benefit from a separate EEP, as there are some extra considerations, as doctesting exceptions, unparseable formats, etc, but it is worth mentioning them given their benefits to users and documentation authors.
+
+EDoc and source code documentation
+==================================
+
+## Should documentation be in the source code?
+
+There is one discussion which is the debate if API Reference documentation should exist in the source code or be placed in a separate file. There are arguments against and in favor of both approaches. If the documentation is alongside the source code, it is easier to keep it up to date. At the same time, it may pollute the source code. This proposal does not advocate that one is better than the other. Rather, because we already have a tool that colocates documentation and source code, namely EDoc, this EEP aims to propose a more integrated variant of said tool, which will hopefully lead developers to better document their Erlang modules and libraries.
+
+## What about EDoc?
+
+If this proposal is to be accepted, what happens with Edoc?
+
+My proposal would be for EDoc to be repurposed with a new backend that translates the existing EDoc documentation to the new module attribute format. Eventually EDoc can be deprecated and phased out.
+
+The HTML rendering part of EDoc can be repurposed to work on the Docs chunk, allowing the Docs chunk to benefit from it. The Build and Packaging Working Group of the Erlang Ecosystem Foundation is also working on getting the ExDoc project ready to generate HTML+EPUB for Erlang projects. The goal is to distribute ExDoc as an escript which will make it really easy to integrate with existing Erlang libraries via a convenient Command Line Interface.
+
+Similarly, if the goal is to eventually phase out EDoc, the Erlang/OTP team needs to discuss what happens with the existing EDoc to XML conversion in `erl_docgen`. Two options exist:
+
+  * Generate EDoc to XML one last time and use the XML as the source from now on
+
+  * Convert EDoc to the documentation style in this proposal and teach erl_docgen how to convert the Docs chunk to XML
+
+Copyright
+=========
+
+This document has been placed in the public domain.
```

Comment:
This probably should strictly describe license, like CC0, as not all jurisdictions allow voluntary assigning document to public domain.

Reply:
This is the standard of all other EEPs, so I won't change it, but perhaps this conversation needs to brought up elsewhere? Or maybe it only matters for what is valid in Sweden in this case?

Reply:
@josevalim created #25

Reply:
AFAIK, the templates were copied from PEP, so this phrasing is likely just an old artefact. I agree it should be changed to CC0. In many countries (including most of Europe) saying "this is placed in the public domain" does not have any legal effect. Copyright is automatic as of the Berne convention of 1988.